app/assets/javascripts/discourse/app/components/glimmer-header.gjs

```diff
@@ -0,0 +1,240 @@
+    switch (msg.type) {
+      case "search":
+        this.toggleSearchMenu();
+        break;
+      case "user":
+        this.toggleUserMenu();
+        break;
+      case "hamburger":
+        this.toggleHamburger();
+        break;
+      case "page-search":
+        if (!this.togglePageSearch()) {
+          msg.event.preventDefault();
+          msg.event.stopPropagation();
+        }
+        break;
+    }
+  }
+
+  @action
+  toggleSearchMenu() {
+    if (this.site.mobileView) {
+      const context = this.search.searchContext;
+      let params = "";
+      if (context) {
+        params = `?context=${context.type}&context_id=${context.id}&skip_context=${this.skipSearchContext}`;
+      }
+
+      if (this.router.currentRouteName === "full-page-search") {
+        scrollTop();
+        document.querySelector(".full-page-search").focus();
+        return false;
+      } else {
+        return DiscourseURL.routeTo("/search" + params);
+      }
+    }
+
+    this.search.visible = !this.search.visible;
+    if (!this.search.visible) {
+      this.search.highlightTerm = "";
+      this.search.inTopicContext = false;
+      document.getElementById(SEARCH_BUTTON_ID)?.focus();
+    }
+  }
+
+  @action
+  togglePageSearch() {
+    this.search.inTopicContext = false;
+
+    let showSearch = this.router.currentRouteName.startsWith("topic.");
+    // If we're viewing a topic, only intercept search if there are cloaked posts
+    if (showSearch) {
+      const controller = this.register.lookup("controller:topic");
+      const total = controller.get("model.postStream.stream.length") || 0;
+      const chunkSize = controller.get("model.chunk_size") || 0;
+      showSearch =
+        total > chunkSize &&
+        document.querySelector(
+          ".topic-post .cooked, .small-action:not(.time-gap)"
+        )?.length < total;
+    }
+
+    if (this.search.visible) {
+      this.toggleSearchMenu();
+      return showSearch;
+    }
+
+    if (showSearch) {
+      this.search.inTopicContext = true;
+      this.toggleSearchMenu();
+      return false;
+    }
+
+    return true;
+  }
+
+  @action
+  toggleUserMenu() {
+    this.header.userVisible = !this.header.userVisible;
+    this.toggleBodyScrolling(this.header.userVisible);
+    this.args.animateMenu();
+
+    // auto focus on first button in dropdown
+    schedule("afterRender", () =>
+      document.querySelector(".user-menu button")?.focus()
+    );
+  }
+
+  @action
+  toggleHamburger() {
+    if (this.args.sidebarEnabled && !this.site.narrowDesktopView) {
+      this.args.toggleSidebar();
+      this.args.animateMenu();
+    } else {
+      this.header.hamburgerVisible = !this.header.hamburgerVisible;
+      this.toggleBodyScrolling(this.header.hamburgerVisible);
+      this.args.animateMenu();
+
+      schedule("afterRender", () => {
+        // Remove focus from hamburger toggle button
+        document.querySelector("#toggle-hamburger-menu")?.blur();
+      });
```

Comment:
As above, could we remove this and put `{{auto-focus}}` on the first focusable element in the hamburger?